src/builders/indexable-author-builder.php

```diff
@@ -168,24 +181,28 @@ protected function find_alternative_image( Indexable $indexable ) {
 	}
 
 	/**
-	 * Returns the timestamps for a given author.
+	 * Returns public post aggregates for a given author.
 	 *
-	 * @param int $author_id  The author ID.
+	 * @param int $author_id The author ID.
 	 *
-	 * @return object An object with last_modified and published_at timestamps.
+	 * @return object  An object with the number of public posts, most recent last modified and first published at timestamps.
 	 */
-	protected function get_object_timestamps( $author_id ) {
+	protected function get_public_post_archive_aggregates( $author_id ) {
 		$post_statuses = $this->post_helper->get_public_post_statuses();
+		$post_types    = $this->author_archive->get_author_archive_post_types();
 
 		$sql = "
-			SELECT MAX(p.post_modified_gmt) AS last_modified, MIN(p.post_date_gmt) AS published_at
+			SELECT 
```

Comment:
idea: how about getting post ID's for an author first and iterating over those - using primary keys seems more efficient than where-ing over a set of fields.